tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py

```diff
@@ -0,0 +1,135 @@
+
+"""
+copyright: Copyright (C) 2015-2022, Wazuh Inc.
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+type: system
+brief: Check that when an agent pointing to a worker node is registered using enrrolment method and with 
+       group the change is sync with the cluster.
+tier: 0
+modules:
+    - cluster
+components:
+    - manager
+    - agent
+path: /tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py
+daemons:
+    - wazuh-db
+    - wazuh-clusterd
+os_platform:
+    - linux
+os_version:
+    - Arch Linux
+    - Amazon Linux 2
+    - Amazon Linux 1
+    - CentOS 8
+    - CentOS 7
+    - CentOS 6
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+references:
+    - https://github.com/wazuh/wazuh-qa/issues/2510
+tags:
+    - cluster
+"""
+
+import os
+import time
+
+import pytest
+from wazuh_testing.tools.system import HostManager
+from system import (check_agent_groups, restart_cluster, clean_cluster_logs,
+                    check_keys_file, remove_cluster_agents, delete_group_of_agents, get_id_from_agent)
+from wazuh_testing.tools import WAZUH_PATH
+
+
+# Hosts
+test_infra_managers = ["wazuh-master", "wazuh-worker1", "wazuh-worker2"]
+test_infra_agents = ["wazuh-agent1"]
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+tmp_path = os.path.join(local_path, 'tmp')
+id_group = 'group_test'
+enrollment_group = """ 
+    <enrollment>
+    <groups>group_test</groups>
+    </enrollment>
+                       """
+
+@pytest.fixture(scope='function')
+def clean_environment():
+
+    clean_cluster_logs(test_infra_agents + test_infra_managers, host_manager)
+
+    yield
+    # Remove the agent once the test has finished
+    remove_cluster_agents(test_infra_managers[0], test_infra_agents, host_manager)
```

Comment:
we only required to call it as a parameter. Remove this method defined from here.